sound/soc/intel/boards/hda_dsp_common.c

```diff
@@ -10,12 +10,14 @@
 
 #include "hda_dsp_common.h"
 
+#if IS_ENABLED(CONFIG_SND_SOC_SOF_HDA_AUDIO_CODEC)
```

Comment:
are there difference with SND_SOC_SOF_HDA_LINK and SND_SOC_SOF_HDA_AUDIO_CODEC? can we run with HDMI when SND_SOC_SOF_HDA_LINK is set while SND_SOC_SOF_HDA_AUDIO_CODEC is not?

Reply:
@keyonjie This is somewhat hard to understand, but yes, there is a difference. In theory you could use the old hdac-hdmi codec driver with HDA_AUDIO_CODEC disabled, but in practise we have no machine drivers upstream that would support this combination (they all depend on SOF_HDA_AUDIO_CODEC). I have a entry in https://github.com/thesofproject/linux/issues/990 to simplify this, but for now @plbossart wants to keep this (and keep it align with SOUNDWIRE which also has a separate LINK config option).